core/file_walker.go

```diff
@@ -1,50 +1,70 @@
 package core
 
 import (
-	"fmt"
+	"log"
 	"os"
 	"path/filepath"
 )
 
 type File struct {
 	Name  string
+	Dir   *File
 	Size  int64
-	IsDir bool
+	IsDir bool // TODO(leon): IsDir() => len(f.Files) > 0
```

Comment:
Sorry, this was not intended to be checked in. You're right, it doesn't make sense to have that function.